lib/node_modules/@stdlib/math/base/special/gamma-delta-ratio/src/main.c

```diff
@@ -0,0 +1,172 @@
+*
+* ```text
+* Copyright John Maddock 2006-7, 2013-14.
+* Copyright Paul A. Bristow 2007, 2013-14.
+* Copyright Nikhar Agrawal 2013-14.
+* Copyright Christopher Kormanyos 2013-14.
+*
+* Use, modification and distribution are subject to the
+* Boost Software License, Version 1.0. (See accompanying file
+* LICENSE or copy at http://www.boost.org/LICENSE_1_0.txt)
+* ```
+*/
+
+#include "stdlib/math/base/special/gamma_delta_ratio.h"
+#include "stdlib/math/base/special/abs.h"
+#include "stdlib/math/base/special/floor.h"
+#include "stdlib/math/base/special/gamma.h"
+#include "stdlib/math/base/special/factorial.h"
+#include "stdlib/constants/float64/max_safe_nth_factorial.h"
+#include "stdlib/math/base/special/gamma_lanczos_sum.h"
+#include "stdlib/math/base/special/log1p.h"
+#include "stdlib/math/base/special/exp.h"
+#include "stdlib/math/base/special/pow.h"
+#include "stdlib/constants/float64/eps.h"
+#include "stdlib/constants/float64/e.h"
+#include "stdlib/constants/float64/gamma_lanczos_g.h"
+
+static const double FACTORIAL_169 = 4.269068009004705e+304;
+
+/**
+* Calculates the ratio of two gamma functions via Lanczos approximation.
+*
+* ## Notes
+*
+* -   When \\( z < \epsilon \\), we get spurious numeric overflow unless we're very careful. This can occur either inside `lanczosSum(z)` or in the final combination of terms. To avoid this, split the product up into 2 (or 3) parts:
+*
+*     ```tex
+*     \begin{align*}
+*     G(z) / G(L) &= 1 / (z \cdot G(L)) ; z < \eps, L = z + \delta = \delta \\
+*     z * G(L) &= z * G(lim) \cdot (G(L)/G(lim)) ; lim = \text{largest factorial}
+*     \end{align*}
+*     ```
+*
+* @private
+* @param z        first gamma parameter
+* @param delta    difference
+* @returns        gamma ratio
+*/
+static double gammaDeltaRatioLanczos( const double z, const double delta ) {
+	double result;
+	double ratio;
+	double zgh;
+
+	if ( z < STDLIB_CONSTANT_FLOAT64_EPS ) {
+		if ( delta > STDLIB_CONSTANT_FLOAT64_MAX_SAFE_NTH_FACTORIAL ) {
+			ratio = gammaDeltaRatioLanczos( delta, STDLIB_CONSTANT_FLOAT64_MAX_SAFE_NTH_FACTORIAL - delta );
+			ratio *= z;
+			ratio *= FACTORIAL_169;
+			return 1.0 / ratio;
+		}
+		return 1.0 / ( z * stdlib_base_gamma( z + delta ) );
+	}
+	zgh = z + STDLIB_CONSTANT_FLOAT64_GAMMA_LANCZOS_G - 0.5;
+	if ( z + delta == z ) {
+		if ( stdlib_base_abs( delta ) < 10.0 ) {
+			result = stdlib_base_exp( ( 0.5 - z ) * stdlib_base_log1p( delta / zgh ) );
+		} else {
+			result = 1.0;
+		}
+	} else {
+		if ( stdlib_base_abs( delta ) < 10.0 ) {
+			result = stdlib_base_exp( ( 0.5 - z ) * stdlib_base_log1p( delta / zgh ) );
+		} else {
+			result = stdlib_base_pow( zgh / ( zgh + delta ), z - 0.5 );
+		}
+		// Split up the calculation to avoid spurious overflow:
+		result *= stdlib_base_gamma_lanczos_sum( z ) / stdlib_base_gamma_lanczos_sum( z + delta );
```

Comment:
@gunjjoshi This has been changed in 1.85.0.

Reply:
L85-L98.

Reply:
I've updated this now.